src/microphasing.rs

```diff
@@ -137,16 +201,23 @@ impl ObservationMatrix {
     }
 
     /// Add read, while considering given interval end. TODO: Think about reads that are not overlapping variant
-    pub fn push_read(&mut self, read: bam::Record, interval_end:u32, interval_start:u32) -> Result<(), Box<Error>> {
+    pub fn push_read(&mut self, read: bam::Record, interval_end:u32, interval_start:u32, reverse: bool) -> Result<(), Box<Error>> {
         let end_pos = read.cigar().end_pos()? as u32;
         let start_pos = read.pos() as u32;
+        debug!("Read Start: {}, Read End: {} - Window Start: {}, Window End {}", start_pos, end_pos, interval_start, interval_end);
         if end_pos >= interval_end && start_pos <= interval_start {
             // only insert if end_pos is larger than the interval end
             let mut obs = Observation { read: read, haplotype: 0 };
             for (i, variant) in self.variants.iter().enumerate() {
                 obs.update_haplotype(i, variant)?;
             }
-            self.observations.entry(end_pos).or_insert_with(|| Vec::new()).push(obs);
+            // debug!("Read: {}; haplotype: {}", String::from_utf8_lossy(obs.read.qname()), obs.haplotype);
+            if reverse {
+                self.observations.entry(start_pos).or_insert_with(|| Vec::new()).push(obs);
```

Comment:
Maybe just infer the pos in the if-else, and have the entry access below, because it is the same code (i.e. less duplication).